modules/finality-verifier/src/lib.rs

```diff
@@ -72,13 +72,28 @@ pub mod pallet {
 		/// The type through which we will verify that a given header is related to the last
 		/// finalized header in our storage pallet.
 		type AncestryChecker: AncestryChecker<<Self::BridgedChain as Chain>::Header, Self::AncestryProof>;
+
+		/// The upper bound on the number of requests allowed by the pallet.
+		///
+		/// Once this bound is reached the pallet will not allow any dispatchables to be called
+		/// until the request count has decreased.
+		#[pallet::constant]
+		type MaxRequests: Get<u32>;
```

Comment:
Since this is a configuration param, perhaps MaxRequestsPerBlock or even MaxSubmittedProofsPerBlock might be more accurate.

Reply:
It's actually not "per block", it's more like `MaxBurstRequests`, cause after this number is reached, as Hernando noted, we throttle the import to one per block.